### ISSUE TITLE
Missing core-js dependency

### ISSUE THREAD

**Issue by medmunds** [particle-iot/particle-cli#491]
When particle-cli 1.41.1 is installed through yarn (or presumably npm), every `particle` command results in `Error: Cannot find module 'core-js/modules/es.array.index-of'`.

The problem appears to be that core-js is listed in devDependencies, so doesn't normally get installed; it should probably be moved to regular dependencies.

To reproduce:

```console
$ node -v
v10.16.0
$ yarn -v
1.16.0
$ yarn add particle-cli
...
success Saved 368 new dependencies.
info Direct dependencies
└─ particle-cli@1.41.1
...
$ particle --version
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'core-js/modules/es.array.index-of'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/Users/medmunds/dev/waterbot/firmware/node_modules/particle-cli/dist/app/cli.js:3:1)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

(And any other `particle` command results in the same error.)

To workaround, install the missing dependency:
```console
$ yarn add core-js
...
success Saved 1 new dependency.
info Direct dependencies
└─ core-js@3.1.3
...
$ particle --version
1.41.1
```

(This was not a problem in particle-cli 1.23.1, so appears to be an injection since then.)

**Comment by busticated** [comment]
thanks for the report :+1:

fixed here: https://github.com/particle-iot/particle-cli/pull/492

**Comment by busticated** [comment]
and released in `v1.41.2` ([changelog](https://github.com/particle-iot/particle-cli/releases/tag/v1.41.2))

reopen if the issue persists :+1:

**Comment by medmunds** [comment]
@busticated thanks, confirmed fixed in v1.41.2.

[Incidentally, if installing with yarn on newer Node versions, yarn's `--ignore-engines` flag is now required due to a [recent change](https://github.com/particle-iot/wiring-preprocessor/commit/5453b2ae09c04ebf976eb2a955f8ee2b76c1b348) in wiring-preprocessor. Unlike npm, yarn is by default strict about package.json `engines` requirements, and wiring-preprocessor now says it wants *only* Node v8. Let me know if you'd like a separate report there.]
